cockatrice/src/gameselector.cpp

```diff
@@ -262,9 +263,11 @@ void GameSelector::actJoin()
         return;
     }
 
-    PendingCommand *pend = r->prepareRoomCommand(cmd);
-    connect(pend, SIGNAL(finished(Response, CommandContainer, QVariant)), this, SLOT(checkResponse(Response)));
-    r->sendRoomCommand(pend);
+    if (!tabSupervisor->switchToGameTabIfAlreadyExists(game.game_id())) {
```

Comment:
the password is still asked for, try moving this check to at least before the password dialog, 18 lines above this one, also try returning immediately, disabling the buttons is not necessary and they will never get enabled again because this happens after when the game gets joined normally.

Reply:
Moved and confirmed no password prompt.

https://user-images.githubusercontent.com/1865818/197104909-df19a83a-2ece-4484-a4fd-aa3b628d4929.mp4